src/ocm/components/AddHosts/utils.ts

```diff
@@ -1,15 +1,32 @@
 import { isSNO } from '../../../common';
-import { canAddHostSNO } from '../../../common/utils';
 import Day2ClusterService from '../../services/Day2ClusterService';
+import {
+  getFeatureSupportLevelsMap,
+  getVersionSupportLevelsMap,
+} from '../featureSupportLevels/FeatureSupportLevelProvider';
 import { OcmClusterType } from './types';
 
 export const canAddHost = ({ cluster }: { cluster: OcmClusterType }) => {
+  const featureSupportLevelsMap =
+    cluster.aiSupportLevels && getFeatureSupportLevelsMap(cluster.aiSupportLevels);
+
+  const versionSupportLevels =
+    featureSupportLevelsMap &&
+    getVersionSupportLevelsMap(
+      cluster.aiCluster?.openshiftVersion || cluster.openshift_version,
+      featureSupportLevelsMap,
+    );
+
+  const isSNOExpansionSupported =
+    versionSupportLevels && versionSupportLevels['SINGLE_NODE_EXPANSION'] === 'supported';
+
   if (Day2ClusterService.getOpenshiftClusterId(cluster)) {
     if (cluster.aiCluster) {
       if (isSNO(cluster.aiCluster)) {
         return (
           cluster.aiCluster.status === 'installed' &&
-          canAddHostSNO(cluster.aiCluster?.openshiftVersion || cluster.openshift_version)
+          cluster.aiCluster.openshiftVersion &&
```

Comment:
Adding this condition makes checking the level support unnecessary when we don't have a `aiCluster.openshiftVersion`, as the results from the validation will be discarded.

I think it can be rewritten as

``` getVersionSupportLevelsMap(
      cluster.aiCluster?.openshiftVersion,
      featureSupportLevelsMap,
    );
```
and there's no need to check the version later.

Reply:
I moved the whole logic elsewhere so hopefully the `canAddHost` function is now cleaner...

Reply:
Thanks, I think it's much clearer now :)